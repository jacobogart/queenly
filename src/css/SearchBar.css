.searchBarContainer {
  border: 1px solid red;
  display: flex;
  height: 40px;
  margin: 0 auto;
  padding: 0.5rem;
  width: 275px;
}

.SearchBar {
  border-radius: 0px 25px 25px 0px;
  border: 1px solid #ccc;
  display: inline-flex;
  flex: 1 1 400px;
  overflow: hidden;
  position: relative;
}

.searchIcon {
  padding: 0.2rem;
  font-size: 16px;
}

.searchTerm {
  border: 1px solid black;
  border: 0;
  flex: 1;
  font-size: 16px;
  padding: 0.2rem 0.2rem 0.2rem 1rem;
}

.searchTerm:focus {
  /* border-radius: 25px 0px 0px 25px; */
  border-right: none;
  /* border: 1px solid black; */
  box-shadow: 0 0 0 3px rgba(0, 123, 255, 0.5);
}

.searchButton:focus {
  outline: none;
}

.searchButton:hover {
  background: #649fdf;
  color: white;
  cursor: pointer;
}

.searchButton {
  background: #538ac5;
  border-radius: 0;
  border: 0;
  color: white;
  padding: 0.5rem;
}
