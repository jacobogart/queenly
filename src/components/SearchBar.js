import React, { Component } from "react";
import SearchSuggestion from "./SearchSuggestion";
import "../css/SearchBar.css";

class SearchBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchQuery: "",
    }
  }

  handleChange = e => {
    if (e.target !== undefined) {
      this.props.updateSearchResults(e.target.value);
      this.setState({
        searchQuery: e.target.value,
      });
    } else {
      this.hideSuggestions();
    }
  }

  findResults = e => {
    e.preventDefault();
    this.props.displayAllSearchResults();
    this.hideSuggestions();
  }

  hideSuggestions = () => {
    this.searchInput.value = '';
    // this.props.hideSearchSuggestions();
  }

  render() {
    return (
      <section
        className="searchBarContainer"
        tabIndex="0"
        onBlur={this.hideSuggestions}
        onMouseLeave={this.props.hideSearchSuggestions}
        ref={el => (this.searchContainer = el)}
      >
        <form onSubmit={this.findResults} className="SearchBar">
          <input
            onChange={this.handleChange}
            ref={el => (this.searchInput = el)}
            type="search"
            placeholder="Search..."
            className="searchTerm"
          />
          <button type="submit" className="searchButton">
            <i className="fas fa-search searchIcon" />
          </button>
        </form>
        <div
          className="searchHolder"
          style={{
            display: this.props.displaySearchSuggestions ? "block" : "none"
          }}
        >
          {this.props.searchResults.slice(0, 5).map(result => (
            <SearchSuggestion
              {...this.props}
              name={result.name}
              selectSearchResult={this.props.selectSearchResult}
              key={result.id}
              hideSuggestions={this.hideSuggestions}
            />
          ))}
        </div>
      </section>
    );
  }
}

export default SearchBar;