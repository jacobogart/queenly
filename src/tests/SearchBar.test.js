import React from 'react';
import SearchBar from '../components/SearchBar';
import { shallow } from 'enzyme';

const mockUpdateResults = jest.fn();
const mockSelectResult = jest.fn();
const mockSearchResults = [
  {
    "bio": "",
    "facebook": "https://www.facebook.com/Sasha-Belle-1457446307864946/",
    "id": 118,
    "imageURL":
      "https://scontent-ort2-1.cdninstagram.com/vp/9e8e535948a6840c9778f3948b3bcd4c/5D2AB39C/t51.2885-19/s320x320/54196427_312652559394918_5887169852254191616_n.jpg?_nc_ht=scontent-ort2-1.cdninstagram.com",
    "instagram": "https://www.instagram.com/sashabelley/?hl=en",
    "name": "Sasha Belle",
    "shows": ["Drag Matinee"],
    "twitter": "https://twitter.com/sashabelle3?lang=en"
  }
];

describe('SearchBar', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <SearchBar updateResults={mockUpdateResults}
                 selectResult={mockSelectResult} 
                 mockSearchResults={mockSearchResults} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state', () => {
    expect(wrapper.state('searchQuery').toEqual(''));
  });

  it('should update searchQuery on change', () => {
    wrapper.find('.searchTerm').simulate('change');

    expect(mockUpdateResults).toBeCalled();
    // expect(wrapper.state('searchQuery').toEqual(''));

    // wrapper.instance().handleChange( {target: {value: 'Sasha'} } );

    // expect(wrapper.state('searchQuery').toEqual('Sasha'))
  });
});




