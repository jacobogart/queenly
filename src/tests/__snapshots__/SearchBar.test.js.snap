// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match snapshot 1`] = `
<section
  className="searchBarContainer"
>
  <form
    className="SearchBar"
    onSubmit={[Function]}
  >
    <input
      className="searchTerm"
      onKeyUp={[Function]}
      placeholder="Search..."
      type="search"
    />
    <button
      className="searchButton"
      type="submit"
    >
      <i
        className="fas fa-search searchIcon"
      />
    </button>
  </form>
  <div
    className="searchHolder"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <SearchSuggestion
      key="118"
      name="Sasha Belle"
      selectSearchResult={[MockFunction]}
    />
  </div>
</section>
`;
